OrbitGl/GlSlider.cpp

```diff
@@ -34,41 +32,76 @@ void GlSlider::SetSliderWidthRatio(float a_WidthRatio)  // [0,1]
   m_Length = std::max(a_WidthRatio, minWidth);
 }
 
-void GlSlider::OnPick(int a_X, int a_Y) {
-  m_Vertical ? OnPickVertical(a_X, a_Y) : OnPickHorizontal(a_X, a_Y);
+Color GlSlider::GetLighterColor(const Color& color) {
+  return Color(static_cast<unsigned char>(color[0] * 1.25),
+               static_cast<unsigned char>(color[1] * 1.25),
+               static_cast<unsigned char>(color[2] * 1.25), 255);
 }
 
-void GlSlider::OnPickHorizontal(int a_X, int /*a_Y*/) {
-  float canvasWidth = m_Canvas->getWidth();
-  float sliderWidth = m_Length * canvasWidth;
-  float nonSliderWidth = canvasWidth - sliderWidth;
+Color GlSlider::GetDarkerColor(const Color& color) {
+  return Color(static_cast<unsigned char>(color[0] * 0.75),
+               static_cast<unsigned char>(color[1] * 0.75),
+               static_cast<unsigned char>(color[2] * 0.75), 255);
```

Comment:
1.25 and 0.75 should probably be constants. Code could maybe be extracted into a function to "scale" a color. It could be used in GetBoxGradientColors as well.

Reply:
Changed to a constant in GlSlider. `GetBoxGradientColors` uses a very different factor though, so I didn't unify this.